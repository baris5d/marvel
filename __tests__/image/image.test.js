import Image from "../../utils/image";
import {shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';

configure({adapter: new Adapter()});

const errorUrl = "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_uncanny.jpg"

const wrapper = shallow(<Image />);

describe("Image Component", () => {
    afterAll(() => wrapper.unmount())

    it("should render a placeholder", () => {
        wrapper.find("img").prop("onError")()
        wrapper.update()

        expect(wrapper.find("img").prop("src")).toBe(errorUrl)

    })

    it("should render an image ", () => {
        const testProps = {
            src:"http://i.annihil.us/u/prod/marvel/i/mg/3/80/4c00358ec7548/portrait_uncanny.jpg",
            alt:"Abyss"
        }
        const wrapper = shallow(<Image {...testProps}/>)

        expect(wrapper.find("img").prop("src")).toBe(testProps.src);
        expect(wrapper.find("img").prop("alt")).toBe(testProps.alt);
    })

    it("renders correctly", () => {
        const testProps = {
            src:"http://i.annihil.us/u/prod/marvel/i/mg/3/80/4c00358ec7548/portrait_uncanny.jpg",
            alt:"Abyss"
        }

        const wrapper = shallow(<Image {...testProps}/>)
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })
})