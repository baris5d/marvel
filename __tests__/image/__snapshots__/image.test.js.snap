// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component renders correctly 1`] = `
<Fragment>
  <img
    alt="Abyss"
    className={
      Array [
        undefined,
        "z-0",
      ]
    }
    onError={[Function]}
    onLoad={[Function]}
    src="http://i.annihil.us/u/prod/marvel/i/mg/3/80/4c00358ec7548/portrait_uncanny.jpg"
  />
  <ContentLoader
    backgroundColor="#374151"
    className="z-10 absolute t-0"
    foregroundColor="#1f2937"
    height={336.75}
    speed={4}
    viewBox="0 0 300 450"
    width={224.5}
  >
    <rect
      height="100%"
      rx="0"
      ry="0"
      width="100%"
      x="0"
      y="0"
    />
  </ContentLoader>
</Fragment>
`;
